source/material_model/drucker_prager.cc

```diff
@@ -0,0 +1,396 @@
+    void
+    DP<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Compositional fields");
+      {
+        // This material model does not incorporate a dependence of material properties
+        // on compositional fields
+        AssertThrow(prm.get_integer ("Number of fields") == 0,
+                    ExcMessage("This material model does not account for compositional fields."));
+      }
+      prm.leave_subsection();
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection("Drucker Prager");
+        {
+          reference_rho              = prm.get_double ("Reference density");
+          reference_T                = prm.get_double ("Reference temperature");
+          reference_eta              = prm.get_double ("Reference viscosity");
+          thermal_k                  = prm.get_double ("Thermal conductivity");
+          reference_specific_heat    = prm.get_double ("Reference specific heat");
+          thermal_alpha              = prm.get_double ("Thermal expansion coefficient");
+          prm.enter_subsection ("Viscosity");
+          {
+            minimum_viscosity = prm.get_double ("Minimum viscosity");
+            maximum_viscosity = prm.get_double ("Maximum viscosity");
+            reference_strain_rate        = prm.get_double ("Reference strain rate");
+            // Convert degrees to radians
+            phi                          = prm.get_double ("Angle internal friction") * numbers::PI/180.0;
+            cohesion                     = prm.get_double ("Cohesion");
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(DP,
+                                   "drucker prager",
+                                   "A material model that has constant values "
+                                   "for all coefficients but the density and viscosity. The defaults for all "
+                                   "coefficients are chosen to be similar to what is believed to be correct "
+                                   "for Earth's mantle. All of the values that define this model are read "
+                                   "from a section ``Material model/Drucker Prager'' in the input file, see "
+                                   "Section~\\ref{parameters:Material_model/Drucker Prager}."
+                                   "\n\n"
+                                   "The viscosity is computed according to the Drucker Prager frictional "
+                                   "plasticity criterion (non-associative) based on a user-defined "
+                                   "Internal angle of frication and cohesion. In 3D: "
+                                   " $yield_strength = \\frac{6 cohesion cos(\\phi)}{\\sqrt(3) (3+sin(\\phi))} +\\ "
```

Comment:
all of the same values apply here as for the discussion on the documentation of the class itself.